EIPS/eip-2029.md

```diff
@@ -0,0 +1,61 @@
+point, the most important counter is the total number of transactions happened, and this counter will be used to populate the nonces of newly created
+non-contract accounts. This way of populating nonce ensures replay protection for accounts that were evicted and then brought back by sending ether to them.
+
+## Motivation
+<!--The motivation is critical for EIPs that want to change the Ethereum protocol. It should clearly explain why the existing protocol specification is inadequate to address the problem that the EIP solves. EIP submissions without sufficient motivation may be rejected outright.-->
+Ethereum currently does not have a special place in the state for tracking state counters such as number of transactions or number of accounts.
+
+## Specification
+<!--The technical specification should describe the syntax and semantics of any new feature. The specification should be detailed enough to allow competing, interoperable implementations for any of the current Ethereum platforms (go-ethereum, parity, cpp-ethereum, ethereumj, ethereumjs, and [others](https://github.com/ethereum/wiki/wiki/Clients)).-->
+Prior to the block A, a contract is deployed with the following code:
+`0x60 0x20 0x60 0x00 0x80 0x80 0x35 0x54 0x90 0x52 0xF3`, which corresponds to this assembly:
+`PUSH1 32 PUSH1 0 DUP1 DUP1 CALLDATALOAD SLOAD SWAP1 MSTORE RETURN`
+Call to this contract accepts one 32-byte argument, `x`, and returns the value of the storage item [`x`].
+
+This contract is deployed using `CREATE2` opcode in such a way that it has the same address on any network.
+
+## Rationale
+<!--The rationale fleshes out the specification by describing what motivated the design and why particular design decisions were made. It should describe alternate designs that were considered and related work, e.g. how the feature is supported in other languages. The rationale may also provide evidence of consensus within the community, and should discuss important objections or concerns raised during discussion.-->
+Two alternative solutions were considered so far:
+1. Extending the structure of the Ethereum state to introduce more fields, and hence change the way the state root is constructed. The main downside of this
+approach is the impact on the software what is currently coupled with the particular way the state root is constructed. Particularly it affects the software
+that deals with merkle proofs derived from the state root.
+2. Extended State Oracle ([EIP-2014](https://eips.ethereum.org/EIPS/eip-2014)). Under such proposal, there will be a precompile contract with standardised interface, capable of returning
+current values of the counters. However, the actual data being returned by such oracle is not explicitly in the state, and is not Merkelised. It means that all the counters need to be added to the snapshots when the snapshot sync is perform, so they still present in the state, but implicitly.
```

Comment:
This is true, however EIP-2014 also states it could be changed into something with a state. To be discussed later anyway.